ir-engine/ir-engine
# IR-4540 changing camera lerping targets and lerp methodology

## Summary
changing infinite approach lerping toward targets in for timed A to B lerp that ends with always being on target after the specified time. Time is set as a const

There is a case for revisiting all cases of the "smootheLerpAlpha" function, as it is being used incorrectly. It is meant to provide a somewhat logarithmic graph but only if you pass it an accumulating value for time. It is being used with deltaTime instead, which is the time since last frame (and performance dependent)...basically meaning that you would have a roughly fixed value from smootheLerpAlpha given a stable framerate, which would never approach 1.

This also no longer lerps each frame from current position/rotation to target pos/rot (meaning it would effectively accelerate the lerp over time rather than have a fixed speed)

## References
[https://tsu.atlassian.net/browse/IR-4540](https://tsu.atlassian.net/browse/IR-4540)

## QA Steps
in studio hit the play button after moving the camera to wherever you want in the viewport (drag left click or hold right click and use wasd to move)

watch as the camera smoothly lerps to the avatar follow position (3rd person over the shoulder by default) and doesn't first travel to the avatar's feet or do anything weird

after lerping finishes in 1 second the camera should follow rotations and movements of the camera/avatar as usual
